### DISCUSSION TITLE
Way to change/modify `entries`

### DISCUSSION THREAD

**Discussion by YoSTEALTH** [axboe/liburing#623] [Q&A]
Any way to change/modify `entries`  of the `io_uring` after `io_uring_queue_init()` has been initialized?

**Comment by axboe** [comment]
It is not, sq/cq ring entries are fixed past initialization.

**Reply by YoSTEALTH** [reply]
Goal is to use `1` io_uring event manager per thread. There is a chance of multiple functions running its own event manager and thus conflicting(in term of wait) with each other. A simple solution to have 1 running ring is by setting an `global ring`. But function-1 might only need say 8-entries and function-2 will need max entries. Thus a need to modify entries.

I was thinking maybe closing old ring, and starting a new. The problem being what happens to waiting events in old ring.

Not sure how monkey patching `oldptr = newptr` using `realloc` at user-end will effect io_uring back-end.
